File: frontend/src/types/telegram.ts
export interface TelegramIntegration {
  id: number
  shopId: number
  chatId: string
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export interface IntegrationStatus {
  enabled: boolean
  chatId: string
  lastSentAt: string | null
  sentCount7d: number
  failedCount7d: number
}

export interface ConnectPayload {
  botToken: string
  chatId: string
  enabled: boolean
}

export interface CreateOrderPayload {
  number: string
  total: number
  customerName: string
}

export interface CreateOrderResponse {
  order: {
    id: number
    number: string
    total: number
    customerName: string
    createdAt: string
  }
  notifyStatus: 'sent' | 'failed' | 'skipped'
}

export interface ApiError {
  error: string
}

File: backend/internal/dto/responses/telegram.go
package responses

import "time"

type TelegramStatus struct {
	Enabled       bool       `json:"enabled"`
	ChatID        string     `json:"chatId"`
	LastSentAt    *time.Time `json:"lastSentAt"`
	SentCount7d   int64      `json:"sentCount7d"`
	FailedCount7d int64      `json:"failedCount7d"`
}

type ConnectTelegram struct {
	ID        uint      `json:"id"`
	ShopID    uint      `json:"shopId"`
	ChatID    string    `json:"chatId"`
	Enabled   bool      `json:"enabled"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}
